### PR TITLE
mp-b2c-guest-cartpitems

### PR DESCRIPTION
**Pull request by kostiantynlitvinov**
positive and negative cases

## PR Description

TBD

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    When I send a POST request:    /guest-carts/${guestCartId}/guest-cart-items    {"data":{"type":"guest-cart-items","attributes":{}}}
+    Then Each array element of array in response should contain property with value:    [errors]    code    901
+    And Each array element of array in response should contain property with value:    [errors]    status    422
+    And Response reason should be:    Unprocessable Content
+    And Array in response should contain property with value:    [errors]    detail    sku => This field is missing.
+    And Array in response should contain property with value:    [errors]    detail    quantity => This field is missing.
+
+Add_an_item_to_the_guest_cart_without_sku_and_quantity_values
+    [Setup]    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    Run Keywords    I send a POST request:    /guest-cart-items    {"data":{"type":"guest-cart-items","attributes":{"sku":"${concrete_available_product_with_offer}","quantity":"1"}}}
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot:61]: for creating a guest cart you should use method 
Create a guest cart:

(here ad in other tests where you do not check actual creation of the cart

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Test Setup        TestSetup
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+Add_an_item_to_the_guest_cart_without_x_anonymous_customer_unique_id
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot:11]: a test is missing for X-Anonymous-Customer-Unique-Id not matching the cart (cart created for another X-Anonymous-Customer-Unique-Id) - add and update items i such cart

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    And Response should return error code:    103
+    And Response reason should be:    Not Found
+    And Response should return error message:    Item with the given group key not found in the cart.
+
+Remove_an_item_from_the_non_existing_guest_cart
+    [Setup]    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    Run Keywords    I send a POST request:    /guest-cart-items    {"data":{"type":"guest-cart-items","attributes":{"sku":"${concrete_available_product_with_offer}","quantity":"1"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    guestCartId
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot:162]: why do we prepare a cart if we do ot use it - we do not need this preparation

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Add_an_item_to_the_guest_cart_with_items_include
+    [Setup]    I set Headers:    Content-Type=${default_header_content_type}    X-Anonymous-Customer-Unique-Id=${random}
+    Run Keywords    I send a POST request:    /guest-cart-items    {"data":{"type":"guest-cart-items","attributes":{"sku":"${concrete_available_product_with_offer}","quantity":"1"}}}
+    ...    AND    Response status code should be:    201
+    ...    AND    Response reason should be:    Created
+    ...    AND    Save value to a variable:    [data][id]    guestCartId
+    ...    AND    I send a DELETE request:    /guest-carts/${guestCartId}/guest-cart-items/${concrete_available_product_with_offer}
+    ...    AND    Response status code should be:    204
+    ...    AND    Response reason should be:    No Content
+    # ...    AND    Cleanup items in the guest cart:    ${guestCartId}
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:51]: why is it commented?

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+Test Setup        TestSetup
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+
+Add_an_item_to_the_guest_cart
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:11]: please add tests with includes for concrete and abstract products
tests with promotional products, cart rules and bundles (you can have a look at this PR for reference https://github.com/spryker/robotframework-suite-tests/pull/120/files

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    # ...    AND    Cleanup items in the guest cart:    ${guestCartId}
+    When I send a POST request:    /guest-carts/${guestCartId}/guest-cart-items?include=items    {"data":{"type":"guest-cart-items","attributes":{"sku":"${concrete_available_product_with_offer}","quantity":"1"}}}
+    Then Response status code should be:    201
+    And Response reason should be:    Created
+    And Response body parameter should be:    [included][0][type]    guest-cart-items
+    And Response body parameter should be:    [included][0][id]    ${concrete_available_product_with_offer}
+    And Response should contain the array of a certain size:    [included]    1
+    And Response body parameter should not be EMPTY:    [data][links][self]
+
+Update_an_item_quantity_at_the_guest_cart_with_items_include
```
Comment (OKrivtsova) [inline tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:60]: this test fails on staging
image

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot (hunk 1 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot:61]: method Create a guest cart is now used for all testcases

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot (hunk 4 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:51]: Cleanup items in the guest cart: not works
Now I'm using new one  - Cleanup All Items In The Guest Cart:

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot (hunk 3 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2c/glue/guest_carts/guest_cart_items/negative.robot:162]: removed

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot (hunk 5 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:11]: includes for concrete and abstract products
tests with promotional products, cart rules and bundles were added

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot (hunk 6 above)
Comment (kostiantynlitvinov) [reply tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:105]: I believe that now it works

File: tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot (hunk 5 above)
Comment (OKrivtsova) [reply tests/api/mp_b2c/glue/guest_carts/guest_cart_items/positive.robot:11]: I did not find tests with cart rules included

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**